trace_events: add version metadata

jasnell opened the pull request:
Use `TRACE_EVENT_METADATA1` to include Node.js, v8, and libuv version
details in the trace log (helpful for log analysis when trace detail
may vary from one Node.js version to another).

/cc @ofrobots 

##### Checklist
- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

jasnell commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/14987/

jasnell commented:
Failures in CI are known flakies

jasnell commented:
Mimic in which way?

ofrobots commented on src/node.cc:
```diff
@@ -4234,6 +4234,11 @@ inline int Start(Isolate* isolate, IsolateData* isolate_data,
   Environment env(isolate_data, context, v8_platform.GetTracingAgent());
   env.Start(argc, argv, exec_argc, exec_argv, v8_is_profiling);
 
+  TRACE_EVENT_METADATA1("__metadata", "version", "node", NODE_VERSION + 1);
```
nit: I'm trying to understand the `+ 1`. If you're skipping the `v` in the version string: 1) why? 2) it would be good to add a comment explaining what the `+ 1` does as it is not immediately obvious. I would prefer if the `v` was included too.

devsnek replied:
i thought we had a lint rule against messing with pointers like that

jasnell replied:
It is for consistency with the value of `process.versions.node` (which does not include the `v`)

cjihrig commented:
> Mimic in which way?

I meant using the same names (for example n-api vs napi), and including the other missing dependencies.

jasnell commented:
> I meant using the same names (for example n-api vs napi), and including the other missing dependencies.

Certainly can do that, yes.

jasnell commented:
Updated .... After talking it over more with @ofrobots ... I removed everything but the node version in this to give us more flexibility later. We need to think about the best approach to handling complex data as opposed to emitting multiple trace events. Emitting the node version covers the most basic use case for this -- that is, knowing what Node.js version generated the trace file -- so it covers the immediate need.

jasnell commented:
New CI: https://ci.nodejs.org/job/node-test-pull-request/15173/

jasnell commented:
Landed in e60eed1
